src/openloco/windows/IndustryWindow.cpp

```diff
@@ -809,6 +562,131 @@ namespace openloco::ui::windows::industry
             { transported::widgets, widx::tab_transported, &transported::events, &common::enabledWidgets }
         };
 
+        static void setDisabledWidgets(window* self)
+        {
+            auto industryObj = objectmgr::get<industry_object>(industrymgr::get(self->number)->object_id);
+            auto disabledWidgets = 0;
+
+            if (industryObj->produced_cargo_type[0] == 0xFF)
+                disabledWidgets |= (1 << common::widx::tab_production);
+
+            if (industryObj->produced_cargo_type[1] == 0xFF)
+                disabledWidgets |= (1 << common::widx::tab_production_2);
+
+            self->disabled_widgets = disabledWidgets;
+        }
+
+        // 0x00456079
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            // Draw Units of Cargo sub title
+            const auto industry = industrymgr::get(self->number);
+            const auto industryObj = objectmgr::get<industry_object>(industry->object_id);
+            const auto cargoObj = objectmgr::get<cargo_object>(industryObj->produced_cargo_type[0]);
+
+            {
+                auto args = FormatArguments();
+                args.push(cargoObj->units_and_cargo_name);
+
+                int16_t x = self->x + 2;
+                int16_t y = self->y - 24 + 68;
+
+                gfx::draw_string_494B3F(*dpi, x, y, colour::black, string_ids::production_graph_label, &args);
+            }
+
+            // Draw Y label and grid lines.
+            int32_t yTick = 0;
+            for (int16_t yPos = self->y + self->height - 7; yPos >= self->y + 68; yPos -= 20)
+            {
+                auto args = FormatArguments();
+                args.push(yTick);
+
+                gfx::draw_rect(dpi, self->x + 41, yPos, self->x + 280, 1, colour::get_shade(self->colours[1], 4));
+
+                gfx::draw_string_494C78(*dpi, self->x + 39, yPos - 6, colour::black, string_ids::population_graph_people, &args);
+
+                yTick += 1000;
+            }
+
+            month_id month = current_month();
+            int16_t year = current_year();
+            int8_t yearSkip = 0;
+
+            for (uint8_t i = industry->history_size[0] - 1; i > 0; i--)
```

Comment:
You will need to choose between history_size[0] and history_size[1] based on the current tab.